vault/data_source_azure_access_credentials.go

```diff
@@ -125,87 +128,84 @@ func azureAccessCredentialsDataSourceRead(d *schema.ResourceData, meta interface
 			return err
 		}
-		config.AADEndpoint = env.ActiveDirectoryEndpoint
-	}
-	authorizer, err := config.Authorizer()
-	if err != nil {
-		return nil
-	}
-	vnetClient.Authorizer = authorizer
 
-	credValidationTimeoutSecs := d.Get("max_cred_validation_seconds").(int)
-	sequentialSuccessesRequired := d.Get("num_sequential_successes").(int)
-	secBetweenTests := d.Get("num_seconds_between_tests").(int)
-
-	startTime := time.Now()
-	endTime := startTime.Add(time.Duration(credValidationTimeoutSecs) * time.Second)
+		switch env.Name {
+		case "AzureChinaCloud":
+			clientOptions.Endpoint = arm.AzureChina
+		case "AzureGovernmentCloud":
+			clientOptions.Endpoint = arm.AzureGovernment
+		case "AzureGermanCloud":
+			// AzureGermanCloud appears to have been removed,
+			// keeping this here to handle the case where the secret engine has
+			// been configured with this environment.
+			clientOptions.Endpoint = arm.Endpoint(env.ResourceManagerEndpoint)
+		}
+	}
 
-	// Please see this data source's documentation for an explanation of the
-	// default parameters used here and why they were selected.
-	sequentialSuccesses := 0
-	overallSuccess := false
+	providerClient := armresources.NewProvidersClient(subscriptionID, creds, clientOptions)
+	ctx := context.Background()
+	delay := time.Duration(d.Get("num_seconds_between_tests").(int)) * time.Second
+	endTime := time.Now().Add(
+		time.Duration(d.Get("max_cred_validation_seconds").(int)) * time.Second)
 	for {
-		if time.Now().After(endTime) {
-			log.Printf("[DEBUG] giving up due to only having %d sequential successes and running out of time", sequentialSuccesses)
-			break
-		}
-		log.Printf("[DEBUG] %d sequential successes obtained, waiting %d seconds to next test client ID and secret", sequentialSuccesses, secBetweenTests)
-		time.Sleep(time.Duration(secBetweenTests) * time.Second)
-
-		// The request we provide here is immaterial because the client is only going to refresh the
-		// token it's using for calls.
-		if _, err := autorest.Prepare(&http.Request{}, vnetClient.WithAuthorization(), vnetClient.WithInspection()); err != nil {
-			// If the creds haven't propagated, we receive an error showing we failed to refresh token.
-			sequentialSuccesses = 0
-			continue
+		pager := providerClient.List(&armresources.ProvidersClientListOptions{
+			Expand: pointerutil.StringPtr("metadata"),
+		})
+
+		for pager.NextPage(ctx) {
+			pr := pager.PageResponse()
+			if pr.RawResponse.StatusCode == http.StatusUnauthorized {
+				return fmt.Errorf("validation failed, unauthorized credentials from Vault, err=%w", pager.Err())
+			}
+			log.Printf("[DEBUG] Provider Client List response %+v", pr)
 		}
-		// If the creds have propagated to the server where we're checking, we receive no error from the above Prepare call.
-		sequentialSuccesses++
-		if sequentialSuccesses == sequentialSuccessesRequired {
-			overallSuccess = true
+
+		if pager.Err() == nil {
+			log.Printf("[DEBUG] Credential validation succeeded")
```

Comment:
With num_sequential_successes being removed this will mean the moment a single Azure AD endpoint returns success then it'll assume that the credential is valid, however there can be a propogation delay between endpoints. with num_sequential_successes this allows for the provider to initiate multiple attempts to ensure credentials are replicated.

Reply:
Interesting, I was not able to reproduce that in my testing with a 5 minute poll. I'm happy to add it back.

Reply:
Is there a way to know which endpoint we hit?

Reply:
I don't believe there is. Replication time seems to take longer when using an the existing SP functionality as documented here [https://www.vaultproject.io/docs/secrets/azure#application-object-ids](https://www.vaultproject.io/docs/secrets/azure#application-object-ids)

Reply:
Fix in https://github.com/hashicorp/terraform-provider-vault/pull/1381/commits/e71e3efc4f2c7e3a715208ac9c37e407620189c6